Make CachedSubimage part of PDFDocumentImage

shallawa opened the pull request:
#### 6f7d211e4d86c1f971e681af47732554e88fd49d
<pre>
Make CachedSubimage part of PDFDocumentImage
<a href="https://bugs.webkit.org/show_bug.cgi?id=252129">https://bugs.webkit.org/show_bug.cgi?id=252129</a>
rdar://105351788

Reviewed by Simon Fraser.

It makes sense for now to have CachedSubimage be part of PDFDocumentImage only.
It can&apos;t be part of BitmapImage since it may display multiple NativeImages. So
in future patches, CachedSubimage will be made part of the NativeImage as well.

* Source/WebCore/platform/graphics/CachedSubimage.cpp:
(WebCore::CachedSubimage::create):
(WebCore::CachedSubimage::createPixelated):
(WebCore::CachedSubimage::createCachedSubimage): Deleted.
(WebCore::CachedSubimage::createPixelatedCachedSubimage): Deleted.
* Source/WebCore/platform/graphics/CachedSubimage.h:
(WebCore::CachedSubimage::imageBuffer const):
(WebCore::CachedSubimage::scaleFactor const):
(WebCore::CachedSubimage::destinationRect const):
(WebCore::CachedSubimage::sourceRect const):
* Source/WebCore/platform/graphics/GraphicsContext.cpp:
(WebCore::GraphicsContext::drawImage):
* Source/WebCore/platform/graphics/Image.cpp:
(WebCore::Image::drawCachedSubimage): Deleted.
(WebCore::Image::destroyDecodedData): Deleted.
* Source/WebCore/platform/graphics/Image.h:
(WebCore::Image::destroyDecodedData):
(WebCore::Image::draw):
(WebCore::Image::cachedSubimageCreateCountForTesting const): Deleted.
(WebCore::Image::cachedSubimageDrawCountForTesting const): Deleted.
(WebCore::Image::drawCachedSubimage): Deleted.
* Source/WebCore/platform/graphics/cg/PDFDocumentImage.cpp:
(WebCore::PDFDocumentImage::createCachedSubimage):
(WebCore::PDFDocumentImage::drawPDFDocument):
(WebCore::PDFDocumentImage::drawFromCachedSubimage):
(WebCore::PDFDocumentImage::draw):
(WebCore::PDFDocumentImage::destroyDecodedData):
* Source/WebCore/platform/graphics/cg/PDFDocumentImage.h:
* Source/WebCore/testing/Internals.cpp:
(WebCore::pdfDocumentImageFromImageElement):
(WebCore::Internals::imageCachedSubimageCreateCount):

Canonical link: <a href="https://commits.webkit.org/260236@main">https://commits.webkit.org/260236@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/760c6db59a17c9e25ec7209147932587aec880a2

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/6/builds/107613 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/77/builds/16667 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/43/builds/40508 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/8/builds/116763 "Built successfully") | [  ~~🛠 wincairo~~](https://ews-build.webkit.org/#/builders/12/builds/116172 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/11/builds/111506 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/76/builds/18089 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/85/builds/7980 "Built successfully") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/36/builds/99784 "Built successfully") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/19/builds/113368 "Passed tests") | [  ~~🧪 ios-wk2~~](https://ews-build.webkit.org/#/builders/78/builds/13654 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/3/builds/96848 "Passed tests") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/35/builds/41322 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/9/builds/95565 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/73/builds/28475 "Passed tests") | [  ~~🧪 api-gtk~~](https://ews-build.webkit.org/#/builders/34/builds/83165 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/81/builds/9656 "Built successfully") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/70/builds/29827 "Passed tests") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/82/builds/10329 "Built successfully") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/84/builds/6725 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/80/builds/15811 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/62/builds/49409 "Passed tests") | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/74/builds/7075 "Built successfully and passed tests") | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/79/builds/11889 "Built successfully") | | | 
| | | | | 
<!--EWS-Status-Bubble-End-->

smfr commented on Source/WebCore/platform/graphics/Image.h:
```diff
@@ -135,7 +134,7 @@ class Image : public RefCounted<Image> {
     virtual String accessibilityDescription() const { return String(); } // null string if unknown
 
-    virtual void destroyDecodedData(bool destroyAll = true);
+    virtual void destroyDecodedData(bool = true) { }
```
I prefer to keep the parameter name, commented out:
`virtual void destroyDecodedData(bool /* destroyAll  */ = true) { }`

shallawa committed:
Make CachedSubimage part of PDFDocumentImage
https://bugs.webkit.org/show_bug.cgi?id=252129
rdar://105351788

Reviewed by Simon Fraser.

It makes sense for now to have CachedSubimage be part of PDFDocumentImage only.
It can't be part of BitmapImage since it may display multiple NativeImages. So
in future patches, CachedSubimage will be made part of the NativeImage as well.

* Source/WebCore/platform/graphics/CachedSubimage.cpp:
(WebCore::CachedSubimage::create):
(WebCore::CachedSubimage::createPixelated):
(WebCore::CachedSubimage::createCachedSubimage): Deleted.
(WebCore::CachedSubimage::createPixelatedCachedSubimage): Deleted.
* Source/WebCore/platform/graphics/CachedSubimage.h:
(WebCore::CachedSubimage::imageBuffer const):
(WebCore::CachedSubimage::scaleFactor const):
(WebCore::CachedSubimage::destinationRect const):
(WebCore::CachedSubimage::sourceRect const):
* Source/WebCore/platform/graphics/GraphicsContext.cpp:
(WebCore::GraphicsContext::drawImage):
* Source/WebCore/platform/graphics/Image.cpp:
(WebCore::Image::drawCachedSubimage): Deleted.
(WebCore::Image::destroyDecodedData): Deleted.
* Source/WebCore/platform/graphics/Image.h:
(WebCore::Image::destroyDecodedData):
(WebCore::Image::draw):
(WebCore::Image::cachedSubimageCreateCountForTesting const): Deleted.
(WebCore::Image::cachedSubimageDrawCountForTesting const): Deleted.
(WebCore::Image::drawCachedSubimage): Deleted.
* Source/WebCore/platform/graphics/cg/PDFDocumentImage.cpp:
(WebCore::PDFDocumentImage::createCachedSubimage):
(WebCore::PDFDocumentImage::drawPDFDocument):
(WebCore::PDFDocumentImage::drawFromCachedSubimage):
(WebCore::PDFDocumentImage::draw):
(WebCore::PDFDocumentImage::destroyDecodedData):
* Source/WebCore/platform/graphics/cg/PDFDocumentImage.h:
* Source/WebCore/testing/Internals.cpp:
(WebCore::pdfDocumentImageFromImageElement):
(WebCore::Internals::imageCachedSubimageCreateCount):

Canonical link: https://commits.webkit.org/260236@main
